src/twisted/words/protocols/irc.py

```diff
@@ -405,6 +405,8 @@ def dataReceived(self, data):
         says CRLF.  (Also, the flexibility of LineReceiver to turn "line mode"
         on and off was not required.)
         """
+        if isinstance(data, bytes):
+            data = data.decode("utf-8")
```

Comment:
This is an important issue, but we could add encoding configurability in a follow-up PR.  Threading this through and providing the APIs to set it for each channel on all the various layers could be a separate issue.